front/src/modules/rollingStock/components/RollingStockEditor/CurveSpreadsheet.tsx

```diff
@@ -77,30 +80,69 @@ const CurveSpreadsheet = ({
         ...(isDefaultCurve ? { default_curve: formattedCurve } : {}),
       },
     };
+
+    // Set the updated effort curves
     setEffortCurves(updatedEffortCurve);
   };
 
-  const orderSpreadsheetValues = () => {
-    const orderedValuesByVelocity = spreadsheetCurve.sort((a, b) => {
-      // if a row has a max_effort, but no speed, it should appear at the top of the table
-      if (b[0] && b[0].value === '') return 1;
-      return Number(a[0]?.value) - Number(b[0]?.value);
-    });
-    updateRollingStockCurve(orderedValuesByVelocity);
-  };
+  const spreadsheetCurve = useMemo(() => {
+    const { speeds, max_efforts } = selectedCurve.curve;
+    let filledDataSheet: DataSheetCurve[] = [];
+
+    if (speeds && max_efforts) {
+      filledDataSheet = max_efforts.map((effort, index) => ({
+        speed: speeds[index] !== undefined ? Math.round(msToKmh(speeds[index]!)) : '',
+        // Effort needs to be displayed in kN
+        effort: effort !== undefined ? effort / 1000 : '',
+      }));
+    }
+
+    // Add an empty line for input only if last line is not already empty
+    if (
+      filledDataSheet.length === 0 ||
+      filledDataSheet[filledDataSheet.length - 1].speed !== '' ||
+      filledDataSheet[filledDataSheet.length - 1].effort !== ''
+    ) {
+      filledDataSheet.push({ speed: '', effort: '' });
+    }
+
+    return filledDataSheet;
+  }, [selectedCurve]);
+
+  useEffect(() => {
+    if (needsSort) {
+      const sortedSpreadsheetValues = spreadsheetCurve
+        // Filter out the last line if speed and effort are not filled
+        .filter((item, index, array) => {
+          if (index === array.length - 1) {
+            return item.speed !== '' && item.effort !== '';
+          }
+          return true;
+        })
```

Comment:
I'm not sure this is very useful, the row will be filtered line 123 no ? :thinking_face:
```suggestion
```